src/main/data/TwoBit.js

```diff
@@ -124,6 +136,32 @@ function markUnknownDNA(basePairs: Array<string>, dnaStartIndex: number, sequenc
   return basePairs;
 }
 
+/**
+ * Try getting a bigger chunk of the remote file
+ * until the Range Error is resolved. This is useful when we need to
+ * parse the header, but when we don't know the size of the header up front.
+ * If the intial request returns an incomplete header and hence the
+ * parsing fails, we next try doubling the requested size.
+ * The size of the request is capped with `untilSize` so that
+ * we don't start asking for MBs of data for no use.
+ * Instead we we throw an error if we reach the cap,
+ * potentially meaning a corrupted 2bit file.
+*/
+function retryRemoteGet(remoteFile: RemoteFile, start: number, size: number, untilSize: number, promisedFunc: Function) {
+  return remoteFile.getBytes(start, size).then(promisedFunc).catch(error => {
+    if(error.constructor == RangeError) {
+      // Do not attempt to download more than `untilSize`
+      if(size > untilSize) {
+        throw `Couldn't parse the header ` +
+              `from the first ${size} bytes of the file. ` +
+              `Corrupted 2bit file?`;
+      }
+      return retryRemoteGet(remoteFile, start, size*2, untilSize, promisedFunc);
```

Comment:
Could you, instead of re-requesting from the `start`, request from `start + size` to `size * 2`, saving some bandwidth and hopefully speeding things up?